src/Client.re

```diff
@@ -1,6 +1,9 @@
 type reply = unit;
 
-type t = Protocol.t;
+type t = {
+  client: Protocol.t,
+  lastRequestId: ref(int),
```

Comment:
What's the reason for tracking this separately for each instance?

Reply:
If I create multiple extension hosts (for example - like the tests do) - the `requestId` should start incrementing from zero for each instance.

Reply:
In addition, it needs to be unique for each instance, because the protocol requires that the `requestId`s are sequential - if we don't do that, we get a message from VSCode saying 'PROTOCOL CORRUPTED'.

Reply:
Ah, thanks for explaining! I still don't think this needs to be tracked in the state though. It just needs the ability to inject an id generator, which you can do as a function parameter. That would even yield more control to the tests, if needed. Having it in the state makes the system inherently more complex, more prone to fail, and because the `ref` isn't encapsulated, any code that has access to it could contain a bug that mutates it and breaks the sequentiality.